lib/util/SparqlExpressionEvaluator.js

```diff
@@ -148,10 +150,10 @@ operators = {
     string = literalValue(string);
     return string.indexOf(substring) >= 0;
   },
-  'regex': function (subject, pattern) {
+  'regex': function (subject, pattern, flag) {
     if (isLiteral(subject))
       subject = literalValue(subject);
-    return new RegExp(literalValue(pattern)).test(subject);
+    return new RegExp(literalValue(pattern), flag).test(subject);
```

Comment:
We might want to check whether the flags are supported. [SPARQL has `smixq`](https://www.w3.org/TR/xpath-functions/#flags). Then again, JavaScript will complain if an invalid flag is passed.